src/detail-page/components/tabs/Tabs.tsx

```diff
@@ -31,60 +31,66 @@ Because it isn't determined how they will look like it's type is unknown.
-      case 'jobs':
-        return <JobsTab content={jobs} />;
-      case 'learn':
-        return <LearnTab content={learn} />;
-      case 'resources':
-        return <ResourcesTab content={resources} />;
-      default:
-        throw new Error('Wrong category name!');
-    }
+const getTabContent = (categoryName: String) => {
+  switch (categoryName) {
+    case 'jobs':
+      return <JobsTab content={tabList.jobs} />;
+    case 'learn':
+      return <LearnTab content={tabList.learn} />;
+    case 'resources':
+      return <ResourcesTab content={tabList.resources} />;
+    default:
+      throw new Error('Wrong category name!');
+  }
+};
+
+export const Tabs: React.FC<TabProps> = ({ categories }) => {
+  const [activeTab, setActiveTab] = useState(0);
+  const { path, url } = useRouteMatch();
+  const history = useHistory();
+  const handleTabClicked = (category: Category, tabIndex: number) => {
+    setActiveTab(tabIndex);
+    history.push(`${url}/${category.name}`);
+  };
+  const syncTabWithPath = (categoryIdFromPath: string) => {
+    const selectedCategoryIndex = categories.findIndex((category) => category.id === categoryIdFromPath);
+    const index = selectedCategoryIndex === -1 ? 0 : selectedCategoryIndex;
+    setActiveTab(index);
   };
 
   return (
-    <StyledCard>
-      <StyledTabs indicatorColor="primary" value={chosenCategory}>
-        {categories.map((category, index) => {
-          const urlPath = index === 0 ? match.url : `${match.url}/${category.name}`;
-          return (
-            <Link to={`${urlPath}`} key={category.name}>
-              <Tab
-                text={category.text}
-                chosenTab={chosenCategory}
-                index={index}
-                onClick={setChosenCategory}
-              />
-            </Link>
-          );
-        })}
-      </StyledTabs>
-      {categories.map((category, index) => {
-        const urlPath = index === 0 ? match.path : `${match.path}/${category.name}`;
-        return (
-          <Route exact path={`${urlPath}`}>
-            <TabPanel key={category.name}>
-              {getTabContent(category.name)}
-            </TabPanel>
-          </Route>
-        );
-      })}
-    </StyledCard>
+    <>
+      <StyledCard>
+        <StyledTabs indicatorColor="primary" value={activeTab}>
+          {categories.map((category, index) => (
+            <Tab
+              key={category.name}
+              text={category.text}
+              chosenTab={activeTab}
+              index={index}
+              handleClick={(tabIndex: number) => handleTabClicked(category, tabIndex)}
+            />
+          ))}
+        </StyledTabs>
+      </StyledCard>
+      <Switch>
+        <Redirect exact from="/details/:professionId" to={`${url}/learn`} />
+        <Route exact path={`${path}/:categoryId`}>
+          <TabPanel
+            getTabContent={getTabContent}
+            syncTabWithPath={syncTabWithPath}
+          />
+        </Route>
+      </Switch>
+    </>
```

Comment:
I have a question here. 

Why our TabPanel is outside StyledCard? Because now our Tabs have this rounded border and some smooth shadow on the bottom. When we will add some content in the next tasks to the TabPanels it will look a little bit weird. I think Tabs and TabPanel content sholud be wrapped in one Card content or some div. But maybe I am missing somoething here.

Reply:
Ah, that's my omission. Fixed (and renamed `StyledCard` to `TabsContainer` to make it a bit more meaningful).